### PR TITLE
2gen

### PR DESCRIPTION
**Pull request by silvaby**
Final.

### CODE REVIEW

File: Problem1.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import UIKit
+
+// MARK: - Create a program to print the mean, median and mode of N numbers.
+// You should enter number N and then numbers, don't use standard functions, write your own
+
+func printMathAverage(enterNumberN numberN: Int, ofNumbers numbers: [Int]) {
+    // MARK: Length check
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Rename. Done.

File: Problem1.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    var arrayForPrint = [Int]()
+    
+    for (index, value) in numbers.enumerated() {
+        if index < numberN {
+            arrayForPrint.append(value)
+        }
+    }
+    
+    // MARK: Mean
+    func printMean() {
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Fixed.

File: Problem1.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        }
+        let mean = sum/Double(arrayForPrint.count)
+        
+        print("The mean of numbers \(arrayForPrint) is \(mean)")
+    }
+    
+    // MARK: Median
+    func printMedian() {
+        var median = 0.0
+        let sortedArray = arrayForPrint.sorted { $0 < $1}
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Thanks.

File: Problem1.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        let midleIndex = sortedArray.count / 2
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Thanks.

File: Problem1.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        
+        if sortedArray.count % 2 == 1 {
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Done.

File: Problem1.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            sum += Double(value)
+        }
+        let mean = sum/Double(arrayForPrint.count)
+        
+        print("The mean of numbers \(arrayForPrint) is \(mean)")
+    }
+    
+    // MARK: Median
+    func printMedian() {
+        var median = 0.0
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Fixed.

File: Problem1.playground/playground.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/playground.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Thank you for the useful information.

File: Problem1.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        } else {
+            median = Double(sortedArray[midleIndex] + sortedArray[midleIndex - 1]) / 2.0
+        }
+        
+        print("The median of numbers \(arrayForPrint) is \(median)")
+    }
+
+    // MARK: Mode
+    func printMode() {
+        var mode = ""
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Ok. Done.

File: Problem1.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    }
+    
+    return mode
+}
+
+// MARK: - Test
+let n = 3
+let zero = 0
+let arrayOfNumbers = [10, 20, 30, 20, 15, 10, 5, 35]
+let empArray = [Int]()
```
Comment (dersim-davaod) [inline Problem1.playground/Contents.swift]: Unused?

nit: I don't like the naming.

What is the principal difference between the `empArray ` and the `emptyArray` below?

File: Problem1.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        let middleIndex = sortedArray.count / 2
+
+        if sortedArray.count.isMultiple(of: 2) {
+            median = Double(sortedArray[middleIndex] + sortedArray[middleIndex - 1]) / 2.0
+        } else {
+            median = Double(sortedArray[middleIndex])
+        }
+    }
+    
+    //print("The median of numbers \(numbers) is \(median)")
```
Comment (dersim-davaod) [inline Problem1.playground/Contents.swift]: Please remove the dead/commented code.

File: Problem1.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+        }
+        
+    }
+    
+    return mode
+}
+
+// MARK: - Test
+let n = 3
+let zero = 0
```
Comment (dersim-davaod) [inline Problem1.playground/Contents.swift]: It doesn't make sense to have such aliases.

File: Problem1.playground/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+// Mean
+func mean(ofNumbers numbers: [Int]) -> Double? {
+    var sum = 0.0
+    var mean: Double?
+    
+    if numbers.isEmpty {
+        mean = nil
+    } else {
+        for value in numbers {
+            sum += Double(value)
```
Comment (dersim-davaod) [inline Problem1.playground/Contents.swift]: Don't do conversion from `Int` to `Double` each time.

As an option, you can have `sum` with type `Int` and you can perform the convertion later:
```swift
mean = Double(sum) / Double(numbers.count)
```

File: Problem1.playground/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+            if index < numberN {
+                arrayForPrint.append(value)
+            }
+        }
+    }
+    
+    return arrayForPrint
+}
+
+// Mean
```
[the remark this answers is not in the document]
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Done.

File: Problem1.playground/Contents.swift (hunk 9 above)
Comment (silvaby) [reply Problem1.playground/Contents.swift]: There is no principle :) This is just for the test.

File: Problem1.playground/Contents.swift (hunk 10 above)
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Ok.

File: Problem1.playground/Contents.swift (hunk 11 above)
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Ok.

File: Problem1.playground/Contents.swift (hunk 12 above)
Comment (silvaby) [reply Problem1.playground/Contents.swift]: Done.

**Comment by silvaby** [comment]
Moved to Task 1 #2 .

**Pull request closed without merging**